src/plone/restapi/services/relations/get.py

```diff
@@ -0,0 +1,326 @@
+                            # type="ImportError",
+                            message=str(e),
+                        )
+                    )
+            else:
+                self.request.response.setStatus(501)
+                return dict(
+                    error=dict(
+                        type="ImportError",
+                        message="Relationhelpers not available. Install collective.relationhelpers or upgrade to Plone 6!",
+                    )
+                )
+
+        # Get broken relations for all relation types
+        if onlyBroken:
+            data = {}
+            relationNames = getBrokenRelationNames()
+            for relationName in relationNames:
+                foo = get_relations(relationship=relationName, onlyBroken=True)
+                data.update(foo)
+            return {
+                "@id": f'{self.request["SERVER_URL"]}{self.request.environ["REQUEST_URI"]}',
+                "items": data,
+                "items_total": dict([(el, len(data[el])) for el in data]),
+            }
+
+        # Stats
+        if not source and not target and not relationship:
+            try:
+                stats = relation_stats()
+                stats[
+                    "@id"
+                ] = f'{self.request["SERVER_URL"]}{self.request.environ["REQUEST_URI"]}'
+                return stats
+            except ImportError:
+                self.request.response.setStatus(501)
+                return dict(
+                    error=dict(
+                        type="ImportError",
+                        message="Relationhelpers not available. Install collective.relationhelpers or upgrade to Plone 6!",
+                    )
+                )
+            except Unauthorized:
+                return self.reply_no_content(status=401)
+
+        # Query relations
+        if source:
+            if source[0:1] == "/":
+                source = plone_api_content_get(path=source)
+            else:
+                source = plone_api_content_get(UID=source)
+            if not source:
+                return self.reply_no_content(status=404)
+            else:
+                sources = [source]
+
+        if target:
+            if target[0:1] == "/":
+                target = plone_api_content_get(path=target)
+            else:
+                target = plone_api_content_get(UID=target)
+            if not target:
+                return self.reply_no_content(status=404)
+            else:
+                targets = [target]
+
+        if query_source:
+            query_objects = {}
+            if query_source[0] == "/":
+                query_objects["path"] = {"query": f"{portal.id}/{query_source}"}
+            else:
+                query_objects["SearchableText"] = query_source
+            results = catalog.searchResults(**query_objects)
+            # TODO Maybe return warning if many results
+            sources = [el.getObject() for el in results]
```

Comment:
In general it might be too easy for this endpoint to keep a backend instance thread busy for long enough that the frontend webserver times out, in a large site. Where are query_source and query_target used in the frontend?

Reply:
This is a real concern, but it goes beyond the scope of implementing this feature, so I will resolve the comment.